src/components/molecules/ImageInput/ImageInput.tsx

```diff
@@ -24,94 +24,82 @@ interface ImageInputProps extends ContainerClassName {
+  image,
+  remoteUrlInputName,
+  remoteImageUrl,
+  containerClassName,
+  imageClassName,
+  name,
+  error,
+  disabled,
+  setValue,
+  register,
+}) => {
+  const imageUrl = useMemo(
+    () =>
+      (image && image.length > 0 && URL.createObjectURL(image[0])) ||
       remoteImageUrl,
-      containerClassName,
-      imageClassName,
-      name,
-      error,
-      disabled,
-      setValue,
-      register,
-    },
-    ref
-  ) => {
-    const imageUrl = useMemo(
-      () =>
-        (image && image.length > 0 && URL.createObjectURL(image[0])) ||
-        remoteImageUrl,
-      [image, remoteImageUrl]
-    );
+    [image, remoteImageUrl]
+  );
 
-    const {
-      loading,
-      errorMessage,
-      handleFileInputChange,
-    } = useImageInputCompression(register, error?.message, name);
+  const {
+    loading,
+    errorMessage,
+    handleFileInputChange,
+  } = useImageInputCompression(register, error?.message, name);
 
-    const handleFileInputChangeWrapper = useCallback(
-      async (event: ChangeEvent<HTMLInputElement>) => {
-        const [url, compressedFile] = await handleFileInputChange(event);
-        if (!compressedFile || !url) return;
+  const handleFileInputChangeWrapper = useCallback(
+    async (event: ChangeEvent<HTMLInputElement>) => {
+      const [url, compressedFile] = await handleFileInputChange(event);
+      if (!compressedFile || !url) return;
 
-        setValue(name, [compressedFile], false);
+      setValue(name, [compressedFile], false);
 
-        if (remoteUrlInputName) {
-          setValue(remoteUrlInputName, url, true);
-        }
-      },
-      [handleFileInputChange, name, remoteUrlInputName, setValue]
-    );
+      if (remoteUrlInputName) {
+        setValue(remoteUrlInputName, url, true);
+      }
+    },
+    [handleFileInputChange, name, remoteUrlInputName, setValue]
+  );
 
-    return (
-      <>
-        <div
-          className={classNames(
-            `image-input default-container`,
-            containerClassName,
-            {
-              disabled: loading,
-            }
-          )}
-        >
-          {imageUrl ? (
-            <img
-              className={`default-image ${imageClassName}`}
-              src={imageUrl}
-              alt="upload"
-            />
-          ) : (
-            <div className="centered-flex empty">
-              <h6 className=" text" style={{ fontWeight: "lighter" }}>
-                Click to upload an image
-              </h6>
-            </div>
-          )}
-          <input
-            disabled={disabled || loading}
-            type="file"
-            accept={ACCEPTED_IMAGE_TYPES}
-            onChange={handleFileInputChangeWrapper}
-            className="default-input"
+  return (
+    <>
+      <div
+        className={classNames(
+          `image-input default-container`,
+          containerClassName,
+          {
+            disabled: loading,
+          }
+        )}
+      >
```

Comment:
Can you please extract this outside and add the block class as well? Something like this should do it

```tsx
    const parentClasses = classNames({
      "ImageInput image-input default-container": true,
      [containerClassName]: true,
      disabled: loading,
    });
```

Reply:
explanation here https://github.com/sparkletown/sparkle/pull/2201#discussion_r700168039